assets/js/googlesitekit/notifications/datastore/notifications.js

```diff
@@ -155,14 +172,104 @@ export const reducer = createReducer( ( state, { type, payload } ) => {
 			break;
 		}
 
+		case RECEIVE_QUEUED_NOTIFICATIONS: {
+			state.queuedNotifications = payload.queuedNotifications;
+			break;
+		}
+
 		default:
 			break;
 	}
 } );
 
-export const resolvers = {};
+export const resolvers = {
+	*getQueuedNotifications( viewContext ) {
+		const registry = yield Data.commonActions.getRegistry();
+
+		const notifications = registry
+			.select( CORE_NOTIFICATIONS )
+			.getNotifications();
+
+		// Wait for all dismissed items to be available before filtering.
+		yield Data.commonActions.await(
+			registry.resolveSelect( CORE_USER ).getDismissedItems()
+		);
+
+		const filteredNotifications = Object.values( notifications ).filter(
+			( notification ) => {
+				if ( ! notification.viewContexts.includes( viewContext ) ) {
+					return false;
+				}
+
+				if (
+					!! notification.isDismissible &&
+					registry
+						.select( CORE_NOTIFICATIONS )
+						.isNotificationDismissed( notification.id )
+				) {
+					return false;
+				}
+
+				return true;
+			}
+		);
+
+		const checkRequirementsResults = yield Data.commonActions.await(
+			Promise.all(
+				filteredNotifications.map( ( notification ) => {
+					if (
+						typeof notification.checkRequirements === 'function'
+					) {
+						return notification.checkRequirements( registry );
```

Comment:
@jimmymadon, one more thing that we need to do here is to make sure that we always return a resolved value because `Promise.all` will be rejected if any of its promises is rejected. In other words, with the current implementation, if one of `checkRequirements` calls is failed and rejected, then all `checkRequirementsResults` will be empty.

We need to do something like this:

```
filteredNotifications.map( async ( { checkRequirements } ) => {
    if ( typeof checkRequirements === 'function' ) {
        try {
            return await checkRequirements( registry );
        } catch ( e ) {
            return false;
        }
    }

    return true;
} )
```

Reply:
Nice catch - this would happen when the `checkRequirements` throws an error rather than returning `false`. Thanks!